src/utils/addressList/index.ts

```diff
@@ -0,0 +1,49 @@
+import { utils as ethersUtils } from "ethers";
+import { AddressListAdapter, INVALID_ADDRESS } from "./types";
+import * as bybit from "./adapters/bybit";
+
+export class AddressList {
+  constructor(readonly addressLists: AddressListAdapter[]) {}
+
+  async update(): Promise<{ length: number; addresses: Set<string> }> {
+    const rawAddresses = await Promise.all(this.addressLists.map((adapter) => adapter.update()));
+    const allAddresses = rawAddresses
+      .flat()
+      .map((address) => {
+        try {
+          return ethersUtils.getAddress(address.toLowerCase());
+        } catch {
+          return INVALID_ADDRESS;
+        }
+      })
+      .filter((address) => address !== INVALID_ADDRESS);
+
+    // Dedup the aggregated, normalised, filtered set of addresses.
+    const addresses = new Set(allAddresses);
+
+    return {
+      addresses,
+      length: Array.from(addresses).length,
```

Comment:
NIT: You can try `addresses.size` on the Set itself to find the length.

Reply:
Damn; I missed that, but that's nice - fixed here: https://github.com/across-protocol/sdk/pull/902/commits/3963dcace03d37594937e499665e46a7aef26e2c